exporter/prometheus/prometheus_test.go

```diff
@@ -174,3 +179,94 @@ func TestCollectNonRacy(t *testing.T) {
 		}
 	}()
 }
+
+type mCreator struct {
+	m   *stats.Int64Measure
+	err error
+}
+
+type mSlice []*stats.Int64Measure
+
+func (mc *mCreator) appendInt64(measures *mSlice, name, desc, unit string) {
+	mc.m, mc.err = stats.Int64(name, desc, unit)
+	*measures = append(*measures, mc.m)
+}
+
+type vCreator struct {
+	v   *view.View
+	err error
+}
+
+func (vc *vCreator) createAndSubscribe(name, description string, keys []tag.Key, measure stats.Measure, agg view.Aggregation) {
+	vc.v, vc.err = view.New(name, description, keys, measure, agg)
+	if err := vc.v.Subscribe(); err != nil {
+		vc.err = err
+	}
+}
+
+func TestMetricsEndpointOutput(t *testing.T) {
+	exporter, err := newExporter(Options{})
+	if err != nil {
+		t.Fatalf("failed to create prometheus exporter: %v", err)
+	}
+	view.RegisterExporter(exporter)
+
+	measures := make(mSlice, 0)
+	mc := &mCreator{}
+	mc.appendInt64(&measures, "tests/foo", "foo", "")
+	mc.appendInt64(&measures, "tests/bar", "bar", "")
+	mc.appendInt64(&measures, "tests/baz", "baz", "")
+	if mc.err != nil {
+		t.Errorf("failed to create measures: %v", err)
+	}
+
+	vc := &vCreator{}
+	for _, m := range measures {
+		vc.createAndSubscribe(m.Name(), m.Description(), nil, m, view.CountAggregation{})
+	}
+	if vc.err != nil {
+		t.Fatalf("failed to create views: %v", err)
+	}
+	view.SetReportingPeriod(time.Millisecond)
+
+	for _, m := range measures {
+		stats.Record(context.Background(), m.M(1))
+	}
+
+	addr := "localhost:9999"
```

Comment:
Let's make a server manually i.e.
```go
import "net/http/httptest"

srv := httptest.NewServer(exporter)
defer srv.Close()

for ... {
    resp, err := http.Get(srv.URL)
}
```

No need for the http.ListenAndServe in the goroutine below.

Reply:
Oooh, nice. Was wondering how else I could that. Thanks!